mikochiku_alarm.py

```diff
@@ -135,45 +150,34 @@ def alarm_sound(self):
-            soup = BeautifulSoup(html.text, 'html.parser')
-            keyword = 'window["ytInitialData"]'
-            for scrp in soup.find_all("script"):
-                if keyword in str(scrp):
-                    dict_str = str(scrp).split(' = ', 1)[1]
-            dict_str = dict_str.replace('false', 'False')
-            dict_str = dict_str.replace('true', 'True')
-
-            index = dict_str.find("\n")
-            dict_str = dict_str[:index-1]
-            dics = eval(dict_str)
-            for section in dics.get("contents", {}).get("twoColumnBrowseResultsRenderer", {}).get("tabs", {})[0].get("tabRenderer", {}).get("content", {}).get("sectionListRenderer", {}).get("contents", {}):
-                for itemsection in section.get("itemSectionRenderer", {}).get("contents", {}):
-                    items = {}
-                    if "shelfRenderer" in itemsection:
-                        for items in itemsection.get("shelfRenderer", {}).get("content", {}).values():
-                            for item in items.get("items", {}):
-                                for videoRenderer in item.values():
-                                    for badge in videoRenderer.get("badges", {}):
-                                        if badge.get("metadataBadgeRenderer", {}).get("style", {}) == "BADGE_STYLE_TYPE_LIVE_NOW":
-                                            video_id_set.add(
-                                                videoRenderer.get("videoId", ""))
-                    elif "channelFeaturedContentRenderer" in itemsection:
-                        for item in itemsection.get("channelFeaturedContentRenderer", {}).get("items", {}):
-                            for badge in item.get("videoRenderer", {}).get("badges", {}):
-                                if badge.get("metadataBadgeRenderer", {}).get("style", "") == "BADGE_STYLE_TYPE_LIVE_NOW":
-                                    video_id_set.add(
-                                        item.get("videoRenderer", {}).get("videoId", ""))
-        except:
+            url = "https://www.youtube.com/channel/" + search_ch_id
+            response = self.session.get(url=url, 
+                headers=HEADERS, timeout=(20,10), stream=True)
+            response.raise_for_status()
+            data = re.search(PATTERN_DATA, response.text)
+            # Find first values of `videoId` following EVERY `LIVE NOW`
+            video_ids = re.findall(PATTERN_LIVE_VIDEO, data.group(1))
+            video_id_set = set(video_ids)
             return video_id_set
+        except requests.exceptions.ConnectTimeout as e:
+            print('タイムアウトしました。')
+            # sessionを張り直して継続する。
+            self.session = self.reset_session()
+            return set()
+        except requests.exceptions.RequestException as e:
+            if response.status_code == 404:
+                print(f'{search_ch_id} は、存在しないチャンネルです。')

```

Comment:
Is there any value supposed to be returned here?
I think `self.session.close()` @ line 176 will be executed if no `return` here, which means a 404 Error will cause the program to abort. Is this expected?

Reply:
>Is there any value supposed to be returned here?

None. 
If a 404 error occurs, the program will stop.
However, I have realized that we are able to change or delete the channel ID in the GUI. Therefore, it is better to notice a warning of invalid channel ID and continue program execution than to abort.

The current implementation does not display the console (pyinstaller --no-console), so I think that it is user-friendly if there is a GUI that pops out errors etc.

Reply:
Thank you for your explanation!